Api/Resource.php

```diff
@@ -50,6 +50,34 @@ class Resource implements ResourceInterface
      */
     private $validationGroups;
     /**
+     * @var bool
+     */
+    private $paginationEnabledByDefault = false;
+    /**
+     * @var bool
+     */
+    private $clientAllowedToEnablePagination = false;
+    /**
+     * @var float
+     */
+    private $itemsPerPageByDefault = 30.;
```

Comment:
This is giving me headaches. Float numbers are approximations and shouldn't be used for things like money values or, as in this case, counting.

You might argue that you are only using the significant, but in reality this will affect other values where you do calculations, which `$itemsPerPage` would most likely be used for.

I strongly vote against using float here.